### PR TITLE
expression:implement vectorized evaluation for `builtinSubstringBinary3ArgsSig`

### PR DESCRIPTION
**Pull request by shihongzhi**
### What problem does this PR solve? 

implement vectorized evaluation for builtinSubstringBinary3ArgsSig , for #12106

### What is changed and how it works?

	$go test -v -benchmem -bench=BenchmarkVectorizedBuiltinStringFunc -run=BenchmarkVectorizedBuiltinStringFunc -args "builtinSubstringBinary3ArgsSig"

	goos: darwin
	goarch: amd64
	pkg: github.com/pingcap/tidb/expression
	BenchmarkVectorizedBuiltinStringFunc/builtinSubstringBinary3ArgsSig-VecBuiltinFunc-4         	   27388	     42875 ns/op	       0 B/op	       0 allocs/op
	BenchmarkVectorizedBuiltinStringFunc/builtinSubstringBinary3ArgsSig-NonVecBuiltinFunc-4      	   10000	    132757 ns/op	       0 B/op	       0 allocs/op
	PASS
	ok  	github.com/pingcap/tidb/expression	3.062s

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by shihongzhi** [comment]
> We should add some test cases about binary string to judge the vec function.

the test case is already in `ast.Substring` at `builtin_string_vec_test.go` file.  But it doesn't get correct message when I run the benchmark command

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -1695,11 +1695,74 @@ func (b *builtinRightBinarySig) vecEvalString(input *chunk.Chunk, result *chunk.
...
 func (b *builtinSubstringBinary3ArgsSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalString evals SUBSTR(str,pos,len), SUBSTR(str FROM pos FOR len), SUBSTR() is a synonym for SUBSTRING().
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_substr
 func (b *builtinSubstringBinary3ArgsSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
```
Comment (qw4990) [inline expression/builtin_string_vec.go]: Please add test case for this function.

Comment (qw4990) [inline expression/builtin_string_vec.go]: Please add a test case that meets [this condition](https://github.com/pingcap/tidb/blob/master/expression/builtin_string.go#L1041).

**Comment by qw4990** [comment]
> > We should add some test cases about binary string to judge the vec function.
> 
> the test case is already in `ast.Substring` at `builtin_string_vec_test.go` file. But it doesn't get correct message when I run the benchmark command

Your test case has to meet this [condition](https://github.com/pingcap/tidb/blob/master/expression/builtin_string.go#L1041) so that this function will be executed.

**Pull request merged**